assets/src/edit-story/utils/useUploadVideoFrame.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../app/api';
+import { useStory } from '../app/story';
+import { useConfig } from '../app/config';
+
+function useUploadVideoFrame( { videoId, src, id } ) {
+	const { actions: { uploadMedia, saveMedia } } = useAPI();
+	const { storyId } = useConfig();
+	const { actions: { updateElementById } } = useStory();
+	const setProperties = useCallback(
+		( properties ) => updateElementById( { elementId: id, properties } ),
+		[ id, updateElementById ] );
+
+	/**
+	 * Returns an image of the first frame of a given video.
+	 *
+	 * @param {string} src Video src URL.
+	 * @return {Promise<string>} The extracted image in base64-encoded format.
+	 */
+	const getFirstFrameOfVideo = useCallback( () => {
+		const video = document.createElement( 'video' );
+		video.muted = true;
+		video.crossOrigin = 'anonymous';
+		video.preload = 'metadata';
+		video.currentTime = 0.5; // Needed to seek forward.
+
+		return new Promise( ( resolve, reject ) => {
+			video.addEventListener( 'error', reject );
+
+			video.addEventListener( 'canplay', () => {
+				const canvas = document.createElement( 'canvas' );
+				canvas.width = video.videoWidth;
+				canvas.height = video.videoHeight;
+
+				const ctx = canvas.getContext( '2d' );
+				ctx.drawImage( video, 0, 0, canvas.width, canvas.height );
+
+				canvas.toBlob( resolve, 'image/jpeg' );
+			} );
+
+			video.src = src;
+		} );
+	}, [ src ] );
+
+	const processData = async () => {
+		try {
+			const obj = await getFirstFrameOfVideo();
+			const { id: featuredMedia, source_url: featuredMediaSrc } = await uploadMedia( obj );
+			await saveMedia( featuredMedia, {
+				meta: {
+					amp_is_poster: true,
```

Comment:
While we're at it, we should change this and `\Google\Web_Stories\Media::POSTER_POST_META_KEY` from `amp_is_poster` to `_web_stories_is_poster` to avoid conflicts

Reply:
Changing it in #140, too.

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/135/commits/a740520f185b95a03f3ec10cc89a9321eebdda05